### ISSUE TITLE
ng build --prod throwing can't resolve all parameters exception

### ISSUE THREAD

**Issue by SamD** [angular/angular-cli#3348]
> Please provide us with the following information:
> ---------------------------------------------------------------

### OS?
> Windows 7, 8 or 10. Linux (which distribution). Mac OSX (Yosemite? El Capitan?)

Linux
cat /etc/lsb-release 
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=16.04
DISTRIB_CODENAME=xenial
DISTRIB_DESCRIPTION="Ubuntu 16.04.1 LTS"

### Versions.
> Please run `ng --version`. If there's nothing outputted, please run in a Terminal: `node --version` and paste the result here:

ng --version
Could not start watchman; falling back to NodeWatcher for file system events.
Visit http://ember-cli.com/user-guide/#watchman for more info.
angular-cli: 1.0.0-beta.19-3
node: 7.2.0
os: linux x64

### Repro steps.
> Was this an app that wasn't created using the CLI? What change did you do on your code? etc.

      "command": "cd client && ng build --target=production --environment=prod",

### The log given by the failure.
> Normally this include a stack trace and some more information.

### Mention any other details that might be useful.

I'm seeing this exception and it appears to be related to zone.js not exactly sure why perhaps someone here might know.
> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

![image](https://cloud.githubusercontent.com/assets/321576/20821209/92363f6c-b7f5-11e6-953a-14c3f01b131f.png)

![image](https://cloud.githubusercontent.com/assets/321576/20821211/9abebcf4-b7f5-11e6-8dbf-8b969ceec8ef.png)

**Comment by SamD** [comment]
Additionally if someone could provide me with a nice way to translate these exceptions from the angular-cli bundle that would be helpful

**Comment by SamD** [comment]
One more thing, in the meantime is there a way to configure angular-cli to produce a bundle in a similar manner to ng serve? I'm not all that familiar with webpack but I would assume all the optimization that goes on in the prod bundle isn't happening with ng serve and as a guess this might be related.

**Comment by SamD** [comment]
Ok based on things working with development ng build I figured this out.

The reason here being that I added in environment.ts an injectable service for configuring things such as enabling debug messages for logging in development. That service was not carried over to environment.prod.ts.

However, the error message generated provided no way me to identify this. I'm not sure if there is something I am missing in my setup that would have made this more obvious but if there is it would be appreciated it someone could provide some hints.
